objects/src/transaction/proven_tx.rs

```diff
@@ -159,6 +125,292 @@ impl ProvenTransaction {
     }
 }
 
+// PROVEN TRANSACTION BUILDER
+// ================================================================================================
+
+/// Builder for a proven transaction.
+#[derive(Clone, Debug)]
+pub struct ProvenTransactionBuilder {
+    /// ID of the account that the transaction was executed against.
+    account_id: Option<AccountId>,
+
+    /// The hash of the account before the transaction was executed.
+    initial_account_hash: Option<Digest>,
+
+    /// The hash of the account after the transaction was executed.
+    final_account_hash: Option<Digest>,
+
+    /// State changes to the account due to the transaction.
+    account_details: Option<AccountDetails>,
+
+    /// List of [Nullifier]s of all consumed notes by the transaction.
+    input_notes: Vec<Nullifier>,
+
+    /// List of [NoteEnvelope]s of all notes created by the transaction.
+    output_notes: Vec<NoteEnvelope>,
+
+    /// State of the output notes.
+    output_note_details: BTreeMap<NoteId, Note>,
+
+    /// The script root of the transaction, if one was used.
+    tx_script_root: Option<Digest>,
+
+    /// Block [Digest] of the transaction's reference block.
+    block_ref: Option<Digest>,
+
+    /// A STARK proof that attests to the correct execution of the transaction.
+    proof: Option<ExecutionProof>,
+}
+
+#[derive(Debug)]
+pub enum ProvenTransactionBuilderError {
```

Comment:
I would rename this into `ProvenTransactionError` and move to `src/errors.rs` module.